components/embedded_form/embedded_form.jsx

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import {
+    checkDialogElementForError,
+    checkIfErrorsMatchElements,
+} from 'mattermost-redux/utils/integration_utils';
+
+import SpinnerButton from 'components/spinner_button';
+import SuggestionList from 'components/suggestion/suggestion_list';
+
+import {localizeMessage} from 'utils/utils.jsx';
+
+import DialogElement from '../interactive_dialog/dialog_element';
+import DialogIntroductionText from '../interactive_dialog/dialog_introduction_text';
+
+export default class EmbeddedForm extends React.PureComponent {
```

Comment:
It looks like most of this component is copied from the `InteractiveDialog` component. Is there an opportunity to refactor the int dialogs into a common component to share code between them, and have the two parents be the difference between the embedded form and int dialog?

Reply:
There is indeed, but until we know the final look of the embedded forms we should wait to unify.